### PR TITLE
Minor improvements on downloadable fonts support

### PR DESCRIPTION
**Pull request by ivaniskandar**
I got frequent crash report on the project I've worked on and it's from the downloadable fonts I used as the AppIntro custom typeface.

```
Fatal Exception: android.content.res.Resources$NotFoundException: Font resource ID #0x7f090001 could not be retrieved.
       at androidx.core.math.MathUtils.clamp(SourceFile:613)
       at androidx.core.math.MathUtils.clamp(SourceFile:489)
       at com.github.paolorotolo.appintro.internal.TypefaceContainer.applyTo(SourceFile:35)
       at com.github.paolorotolo.appintro.AppIntroBaseFragment.onCreateView(SourceFile:113)
       at androidx.fragment.app.Fragment.performCreateView(SourceFile:2539)
       at androidx.fragment.app.FragmentManagerImpl.moveToState(SourceFile:875)
       at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(SourceFile:1227)
       ...
```

I thought the error doesn't make sense since I've used downloadable fonts and it only happens on some devices. So I tried to change the blocking method of `ResourcesCompat.getFont` on `TypefaceContainer` to the non-blocking one. So far I got no more crash report. I also updated the example to use all possible methods to use custom typeface on this library.

### CODE REVIEW

**Comment by Wiktorl4z** [comment]
Hi, I was trying to set custom font using "Support Library compatibility version" but it doesn't work.
Tried few times, example code:
`  var typeface = ResourcesCompat.getFont(this, R.font.customFontName)`  
 `  sliderPage1.titleTypeface = typeface.toString`

and method titleTypefaceFontRes doesn't exist.

I had to update to "AndroidX Capable version" and here it's working fine.
I don't know it it's supposed to work for only newest verion, but decided to let you guys know ;)

File: appintro/src/main/java/com/github/paolorotolo/appintro/internal/CustomFontCache.kt [hunk 1]
```diff
@@ -12,21 +13,20 @@ internal object CustomFontCache {
     private val TAG = LogHelper.makeLogTag(CustomFontCache::class)
     private val cache = hashMapOf<String, Typeface>()
 
-    operator fun get(path: String?, ctx: Context): Typeface? {
+    operator fun get(ctx: Context, path: String?, fontCallback: ResourcesCompat.FontCallback) {
```
Comment (cortinico) [inline appintro/src/main/java/com/github/paolorotolo/appintro/internal/CustomFontCache.kt]:
```suggestion
    fun getFont(ctx: Context, path: String?, fontCallback: ResourcesCompat.FontCallback)
```

Please rename to `.getFont` and also remove the `operator` keyword.
Having a `get` operator will make the `[]` syntax really confusing. Also having `.getFont` will make the naming consistent with `ResourceCompat`.

File: example/src/main/java/com/amqtech/opensource/appintroexample/ui/mainTabs/intro/DefaultIntro.java [hunk 2]
```diff
@@ -25,20 +25,26 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         sliderPage1.setDescription("This is a demo of the AppIntro library.");
         sliderPage1.setImageDrawable(R.drawable.ic_slide1);
         sliderPage1.setBgColor(Color.TRANSPARENT);
+        sliderPage1.setTitleTypefaceFontRes(R.font.lato);
```
Comment (cortinico) [inline example/src/main/java/com/amqtech/opensource/appintroexample/ui/mainTabs/intro/DefaultIntro.java]: As @paolorotolo already mentioned, please move to `DefaultIntro2` or CustomLayout/Background as we want to keep `DefaultIntro` streamlined.

**Comment by ivaniskandar** [comment]
I'm sorry @cortinico I totally forgot I have this PR open. The requested changes is done, but I'm still not quite sure where to put the custom typeface, so I just moved them all to `DefaultIntro2`.

BTW looks like CI is failing because of a bug on detekt. I confirmed it's fixed with latest RC16, other than that I believe it's all good.

**Comment by Wiktorl4z** [comment]
Hi I got similar problem. Happened only on device Android 5.0.2. Would you like to help?

_Caused by android.content.res.Resources$NotFoundException
Font resource ID #0x7f090000 could not be retrieved._
`androidx.core.content.res.ResourcesCompat.getDrawable (ResourcesCompat.java)
androidx.core.content.res.ResourcesCompat.getDrawable (ResourcesCompat.java)
pl.futuredev.tibia.ui.activities.IntroActivity.onCreate (IntroActivity.java)
android.app.Activity.performCreate (Activity.java:5937)
android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1105)
android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2251)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2360)
android.app.ActivityThread.access$800 (ActivityThread.java:144)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1278)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:135)
android.app.ActivityThread.main (ActivityThread.java:5221)
java.lang.reflect.Method.invoke (Method.java)
java.lang.reflect.Method.invoke (Method.java:372)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:899)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:694)`


My code:

`val sliderPage1 = SliderPage()
  sliderPage1.titleTypefaceFontRes = R.font.mystery_quest
  sliderPage1.description = ""
  sliderPage1.imageDrawable = R.drawable.tibiamonsters_logo
  addSlide(AppIntroFragment.newInstance(sliderPage1))`

Font xml
`<?xml version="1.0" encoding="utf-8"?>
<font-family xmlns:app="http://schemas.android.com/apk/res-auto"
        app:fontProviderAuthority="com.google.android.gms.fonts"
        app:fontProviderPackage="com.google.android.gms"
        app:fontProviderQuery="Mystery Quest"
        app:fontProviderCerts="@array/com_google_android_gms_fonts_certs">
</font-family>
`

**Comment by ivaniskandar** [comment]
> Hi I got similar problem. Happened only on device Android 5.0.2. Would you like to help?
> 
> _Caused by android.content.res.Resources$NotFoundException Font resource ID #0x7f090000 could not be retrieved._
> `androidx.core.content.res.ResourcesCompat.getDrawable (ResourcesCompat.java) androidx.core.content.res.ResourcesCompat.getDrawable (ResourcesCompat.java) pl.futuredev.tibia.ui.activities.IntroActivity.onCreate (IntroActivity.java) android.app.Activity.performCreate (Activity.java:5937) android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1105) android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2251) android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2360) android.app.ActivityThread.access$800 (ActivityThread.java:144) android.app.ActivityThread$H.handleMessage (ActivityThread.java:1278) android.os.Handler.dispatchMessage (Handler.java:102) android.os.Looper.loop (Looper.java:135) android.app.ActivityThread.main (ActivityThread.java:5221) java.lang.reflect.Method.invoke (Method.java) java.lang.reflect.Method.invoke (Method.java:372) com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:899) com.android.internal.os.ZygoteInit.main (ZygoteInit.java:694)`
> 
> My code:
> 
> `val sliderPage1 = SliderPage() sliderPage1.titleTypefaceFontRes = R.font.mystery_quest sliderPage1.description = "" sliderPage1.imageDrawable = R.drawable.tibiamonsters_logo addSlide(AppIntroFragment.newInstance(sliderPage1))`
> 
> Font xml
> `<?xml version="1.0" encoding="utf-8"?> <font-family xmlns:app="http://schemas.android.com/apk/res-auto" app:fontProviderAuthority="com.google.android.gms.fonts" app:fontProviderPackage="com.google.android.gms" app:fontProviderQuery="Mystery Quest" app:fontProviderCerts="@array/com_google_android_gms_fonts_certs"> </font-family> `

Yeah it needs this PR to be merged. Any news? @cortinico
I personally just use my fork while waiting, you can do the same.
```
implementation 'com.github.ivaniskandar:AppIntro:-SNAPSHOT'
```

**Comment by Wiktorl4z** [comment]
OK I will add your fork instead. Thanks

**Comment by paolorotolo** [comment]
Ok, since Nicola has been very busy recently and this fixes an issue in master, I'm merging this now.
Thanks for your contribute.

**Comment by cortinico** [comment]
Thanks for doing so @paolorotolo <3
I gave a quick check and all was fine for me anyway :) Looking forward to publishing AppIntro 6.x soon

**Pull request merged**